src/ext/java/org/opentripplanner/ext/flex/trip/ScheduledDeviatedTrip.java

```diff
@@ -109,7 +110,7 @@ public ScheduledDeviatedTrip(Trip trip, List<StopTime> stopTimes) {
     ArrayList<FlexEgressTemplate> res = new ArrayList<>();
 
     for (int fromIndex = toIndex; fromIndex >= 0; fromIndex--) {
-      if (stopTimes[fromIndex].pickupType == NONE.getGtfsCode()) continue;
+      if (!stopTimes[fromIndex].pickupType.isRoutable()) continue;
```

Comment:
```suggestion
      if (!getBoardRule(fromIndex).isRoutable()) continue;
```
If `not routable` is a common use case, then maybe we should add methods for that as well. If this is the only place, then this is the best. Normally I would access class fields directly inside a class, but in this case there is a bit logic in the access - looking up the index, so then it become better to keep that in one place - in the `getBoardRule`.

Reply:
Yes, it makes the code much easier to read. I added the `PickDrop#isNotRoutable` method